### DISCUSSION TITLE
Just wondering: any difference in sqlalchemy between postgres v12 and postgres v16?  Postgres upgrade required code change.

### DISCUSSION THREAD

**Discussion by taigit** [sqlalchemy/sqlalchemy#10420] [Usage Questions]
I'm just curious: 
my code is running sqlalchemy 2.0 all along w/o problems.  I migrated to 2.0 a while ago.

Now, I've upgraded my db:
I was on postres v12.
Then I upgraded my postgres to v16.
My code broke.

I was using ORM sessions using the python 'with' context and I did not do an explicit commit before exiting the context.
This worked in pg12.
In pg16, these sessions seemed to be not commited.
So, I explicitly committed them and my code is back working.

I'm just curious as to why this happened.  It feels like a sqlalchemy v1.4 --> v2.0 type of change, but not a postgres db change.

**Comment by CaselIT** [comment]
Hi,

This sounds like you had your dbapi set to autocommit mode before. I don't know if you can set postgresql to default to autocommit mode.
By default sqlalchemy does not commit when existing a `with` block, unless you started it using `.begin()`

**Comment by taigit** [comment]
> By default sqlalchemy does not commit when existing a with block, unless you started it using .begin()

yup, I'm using  .begin()  in my  'with' contexts

**Reply by CaselIT** [reply]
Then it's strange that you have to also manually commit.
Do you have an example of code that works on 12 and not on 16?

**Comment by zzzeek** [comment]
it would be extremely unusual if something related to client transaction control changed over PG versions 12 to 16 and it's much more likely that in the transition between databases, a change also occurred within the client side of your applicaiton, like DBAPI driver or SQLAlchemy version.
